msprime/provenance.py

```diff
@@ -68,3 +76,108 @@ def get_environment():
     if _environment is None:
         _environment = _get_environment()
     return _environment
+
+
+class ProvenanceEncoderDecoder(json.JSONEncoder):
+    """
+      Extension of the `json` encoder that serializes arbitrary python class objects
```

Comment:
Nitpick: we normally indent these like
```python
def something():
    """
    A docstring describing something
    """
```
happy to change this convention if you like (but it's a lot of code to update!)

Reply:
I prefer your way as it gives more room on the line - just have to change my editor setup!

Have you thought of using https://pypi.org/project/flake8-docstrings/ to enforce aspects of docstrings like this?

Reply:
> Have you thought of using https://pypi.org/project/flake8-docstrings/ to enforce aspects of docstrings like this?

No, but we definitely should! (Across the board for tskit projects too, this would be great to have).